saleor/graphql/giftcard/schema.py

```diff
@@ -0,0 +1,38 @@
+import graphene
+from graphql_jwt.decorators import permission_required
+
+from ...giftcard import models
+from ..core.fields import PrefetchingConnectionField
+from .mutations import (
+    GiftCardActivate,
+    GiftCardCreate,
+    GiftCardDeactivate,
+    GiftCardUpdate,
+    GiftCardVerify,
+)
+from .resolvers import resolve_gift_card
+from .types import GiftCard
+
+
+class GiftCardQueries(graphene.ObjectType):
+    gift_card = graphene.Field(
+        GiftCard,
+        id=graphene.Argument(graphene.ID, required=True),
+        description="Lookup a gift card by ID.",
+    )
+    gift_cards = PrefetchingConnectionField(GiftCard, description="List of gift cards")
+
+    def resolve_gift_card(self, info, **data):
```

Comment:
I think this should be accessible to admins only. Authenticated users will be able to get their gift cards from the `User` type.